lib/matplotlib/legend_handler.py

```diff
@@ -255,6 +256,62 @@ def create_artists(self, legend, orig_handle,
         return [legline, legline_marker]
 
 
+class HandlerLine2D(HandlerNpoints):
+    """
+    Handler for `.Line2D` instances.
+
+    A class similar to the original handler for `.Line2D` instances but
+    that uses a monolithic artist rather than using one artist for the
+    line and another one for the marker(s).  NB: the former handler, in
+    use before Matplotlib 3, is still available as `.HandlerLine2DCompound`.
+
+    """
+    def __init__(self, marker_pad=0.3, numpoints=None, **kw):
+        """
+        Parameters
+        ----------
+        marker_pad : float
+            Padding between points in legend entry.
+
+        numpoints : int
+            Number of points to show in legend entry.
+
+        Notes
+        -----
+        Any other keyword arguments are given to `HandlerNpoints`.
+        """
+        HandlerNpoints.__init__(self, marker_pad=marker_pad,
+                                numpoints=numpoints, **kw)
+
+    def create_artists(self, legend, orig_handle,
+                       xdescent, ydescent, width, height, fontsize,
+                       trans):
+
+        xdata, xdata_marker = self.get_xdata(legend, xdescent, ydescent,
+                                             width, height, fontsize)
+
+        markevery = None
+        if self.get_numpoints(legend) == 1:
+            # Special case: one wants a single marker in the center
+            # and a line that extends on both sides. One will use a
+            # 3 points line, but only mark the #1 (i.e. middle) point.
+            xdata = np.linspace(xdata[0], xdata[-1], 3)
+            markevery = [1]
+
+        ydata = ((height - ydescent) / 2.) * np.ones(xdata.shape, float)
```

Comment:
`np.full_like(xdata, (height - ydescent) / 2)`

Reply:
I'll do that tomorrow ;). (TBH, this was just copy-pasted from the original handler and I did not really look about how to optimize the existing code.)

Reply:
@anntzer Well actually this will likely have to go in a future PR ^^ : `np.full_like` seems to have been introduced with Numpy 1.8 ([release notes](https://www.numpy.org/devdocs/release.html#new-functions-full-and-full-like)) and we still support Numpy 1.7.1, do we not?

Reply:
Master requires 1.10 (see `__version__numpy__` in `__init__.py`).

Reply:
My bad, I looked at the official current doc :sheep:... Should I update `HandlerLineCollection`, `HandlerLine2DCompound`, and `HandlerErrorBar` as well, or would it be better to do so in a separate PR?

Reply:
A separate PR seems fine.

Reply:
I'll open a separate PR to use `np.full_like` in the other relevant handler classes if this PR gets merged. (I am a bit afraid of the conflicts with the original `LineHandler2D` if I do it right now.)